lib/checksum.js

```diff
@@ -16,14 +16,27 @@
 
 'use strict';
 
-const _ = require('lodash');
-const Bluebird = require('bluebird');
+const crypto = require('crypto');
 const sliceStream = require('slice-stream2');
 const progressStream = require('progress-stream');
 const CRC32Stream = require('crc32-stream');
-const DevNullStream = require('dev-null-stream');
 const ImageReadStream = require('./image-read-stream');
-const utils = require('./utils');
+const Bluebird = require('bluebird');
+
+/**
+ * @summary Get a hash stream
+ * @function
+ * @private
+ *
+ * @param {String} algorithm - either `crc32` or anything supported by `crypto.Hash`
+ * @returns {Stream.Transform}
+ */
+const getHashStream = (algorithm) => {
+  if (algorithm === 'crc32') {
+    return new CRC32Stream();
+  }
+  return crypto.createHash(algorithm);
```

Comment:
I get:

```sh
> crypto.createHash('foobar')
Error: Digest method not supported
    at Error (native)
    at new Hash (crypto.js:49:18)
    at Object.Hash (crypto.js:48:12)
    at repl:1:8
    at REPLServer.defaultEval (repl.js:272:27)
    at bound (domain.js:280:14)
    at REPLServer.runBound [as eval] (domain.js:293:12)
    at REPLServer.<anonymous> (repl.js:441:10)
    at emitOne (events.js:101:20)
    at REPLServer.emit (events.js:188:7)
```

It might make sense to provide a more user friendly message indeed!